### PR TITLE
Drop support of PowerShell 3, 4, 6.2, and unsupported 7

### PR DESCRIPTION
**Pull request by nohwnd**
## PR Summary

Remove testing on unsupported versions of Windows PowerShell 3, 4 and PowerShell 6.2, so we only support 5.1, 7.2 and newer based on their lifetimes.

Update TFMs to reflect the supported .NET and .NET Framework versions. Use non-vulnerable packages.

Use embedded PDBs.

## PR Checklist

- [x] PR has meaningful title
- [x] Summary describes changes
- [x] PR is ready to be merged
  - If not, use the arrow next to `Create Pull Request` to mark it as a draft. PR can be marked `Ready for review` when it's ready.
- [x] Tests are added/update *(if required)*
- [ ] Documentation is updated/added *(if required)* <- Not yet, that will be a whole other PR to docs after this is fully released.

### CODE REVIEW

File: src/Pester.psd1 [hunk 1]
```diff
@@ -66,6 +66,33 @@
...
+        'Assert-NotNull'
+        'Assert-NotSame'
+        'Assert-NotType'
+        'Assert-Null'
+        'Assert-Same'
+        'Assert-Type'
+        'Assert-Like'
+        'Assert-NotLike'
+        'Assert-StringEqual'
+        'Assert-StringNotEqual'
```
Comment (nohwnd) [inline src/Pester.psd1:93]:
```suggestion
```

File: src/Pester.psd1 [hunk 2]
```diff
@@ -66,6 +66,8 @@
         'New-PesterContainer'
         'New-PesterConfiguration'
 
+
+
```
Comment (nohwnd) [inline src/Pester.psd1:70]:
```suggestion
```

**Comment by fflaten** [comment]
Should we let this stabilize for a patch or minor release before starting cleanup of PS 3/4 logic and syntax (like `New-Object` -> `::new()`)

**Comment by nohwnd** [comment]
This was reverted in 5.6.0, but was reapplied on main for 6.0.0 and newer.

**Pull request merged**